library/pkparse.c

```diff
@@ -869,6 +877,57 @@ static int pk_parse_key_pkcs1_der(mbedtls_rsa_context *rsa,
 #endif /* MBEDTLS_RSA_C */
 
 #if defined(MBEDTLS_ECP_C)
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+/*
+ * Helper function for deriving a public key from its private counterpart by
+ * using PSA functions.
+ */
+static int pk_derive_public_key(mbedtls_ecp_group *grp, mbedtls_ecp_point *Q,
+                                const mbedtls_mpi *d)
+{
+    psa_status_t status, destruction_status;
+    psa_key_attributes_t key_attr = PSA_KEY_ATTRIBUTES_INIT;
+    size_t curve_bits;
+    psa_ecc_family_t curve = mbedtls_ecc_group_to_psa(grp->id, &curve_bits);
+    /* This buffer is used to store the private key at first and then the
+     * public one (but not at the same time). Therefore we size it for the
+     * latter since it's bigger. */
+    unsigned char key_buf[MBEDTLS_PSA_MAX_EC_PUBKEY_LENGTH];
+    size_t key_len = PSA_BITS_TO_BYTES(curve_bits);
+    mbedtls_svc_key_id_t key_id = MBEDTLS_SVC_KEY_ID_INIT;
+    int ret;
+
+    psa_set_key_type(&key_attr, PSA_KEY_TYPE_ECC_KEY_PAIR(curve));
+    psa_set_key_usage_flags(&key_attr, PSA_KEY_USAGE_EXPORT);
+
+    ret = mbedtls_mpi_write_binary(d, key_buf, key_len);
```

Comment:
This doesn't seem appropriate. Instead of relying on the raw ECP functions, which understand all the corner cases, now you're relying on raw MPI functions, which don't. In particular, this line is not correct for Montgomery curves, see https://github.com/Mbed-TLS/mbedtls/blob/6942cc3da727e0eeaa1aed24fdc9257dde298382/library/ecp.c#L3306-L3326

Reply:
Ah, good point, we should be using `mbedtls_ecp_write_key()` here indeed. (At least for now, then we'll have to find another approach [at some point](https://github.com/orgs/Mbed-TLS/projects/1?card_filter_query=is%3Aopen#column-19391691) but let's proceed step by step.) Can I assume you'll be fixing that in your PR?